syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.edge2lorarpc";
option java_outer_classname = "Edge2LoRaRPCProto";
option objc_class_prefix = "E2L";

package edge2applicationserver;

service Edge2ApplicationServer {
    rpc new_data (NewDataRequest) returns (NewDataResponse);

    rpc store_e2gw_pub_info (e2gw_pub_info) returns (ResponseMessage);
}

message ResponseMessage {
    sint32 status_code = 1;
    bytes message = 2;
}

message e2gw_pub_info {
    string gw_ip_addr = 1;
    string gw_port = 2;
    bytes e2gw_pub_key = 3;
}

message NewDataRequest {
   string name = 1;
   uint64 timetag = 2;
}

message NewDataResponse {
    string message = 1;
}